packages/dev/core/src/Meshes/abstractMesh.ts

```diff
@@ -2794,6 +2887,21 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
     public getConnectedParticleSystems(): IParticleSystem[] {
         return this._scene.particleSystems.filter((particleSystem) => particleSystem.emitter === this);
     }
+
+    /**
+     * Compute a world space hotspot position
+     * @param hotSpotQuery point indices and barycentric
+     * @param res output world position
+     */
+    public getHotSpotToRef(hotSpotQuery: HotSpotQuery, res: Vector3): void {

```

Comment:
different file would be better if possible :-)

Reply:
Any example in the engine I should follow?

Reply:
David created some module level functions in ray.core recently, but basically we're just saying have something like `abstractMesh.functions.ts` (not sure about the name, @sebavan, @RaananW) with:
```ts
export function GetTransformedPositionToRef(mesh: AbstractMesh, index: number, res: Vector3): boolean {
  // same impl as you have, but using mesh instead of this
}

export function GetHotSpotToRef(mesh: AbstractMesh, hotSpotQuery: HotSpotQuery, res: Vector3): void {
  // same impl as you have, but using mesh instead of this
}
```

Reply:
code moved to abstractMesh.hotSpot.ts